File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/TitlePathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class TitlePathType extends PermalinkPathType {

    @Inject
    public TitlePathType() {
        super(100);
    }

    @Override
    public boolean acceptsKey(OrchidPage page, String key) {
        return key.equals("title");
    }

    @Override
    public String format(OrchidPage page, String key) {
        return page.getTitle();
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathTypes/TitlePathType.kt
package com.eden.orchid.posts.permalink.pathTypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.posts.permalink.PermalinkPathType

import javax.inject.Inject

class TitlePathType @Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "title"
    }

    override fun format(page: OrchidPage, key: String): String? {
        return page.title
    }

}

